### DISCUSSION TITLE
install by homebrew as cask ?

### DISCUSSION THREAD

**Discussion by Louis-Fiacre** [open-webui/open-webui#5424] [General]
Hi, I would love to be able to **install open-webui with homebrew**, like `brew install openwebui --cask` . Do you think it could be useful as an alternative of docker/kubernete or classic git clone ?
I'm not a programmer, I quickly read the documentation to create a cask, I don't realize if it's a big job or not. https://docs.brew.sh/Adding-Software-to-Homebrew

Best regards,

**Comment by Trigas** [comment]
i have ollama as a services but had to instal open-webui with pipx .. it works but would prefer a cleaner solution

**Comment by ylluminate** [comment]
Agreed, an official method for installation and integration with Homebrew ollama would be fantastic and help keep the environment cohesive.

Generating a Homebrew formula is not horrible hard with the use of an LLM at present, but it would be preferable to see a solution from the source.

**Comment by dopeytree** [comment]
Any updates on this?

**Comment by selcukduman** [comment]
Not the same thing, but I use uv to keep the environment up-to-date and created a macos shortcurt to start open-webui on system start without requiring a terminal. The command I added to the shortcut app is:

DATA_DIR=~/.open-webui uvx --python 3.11 open-webui@latest serve

uv automatically downloads the updates before starting the app, so it kinda works for me as a workaround until a homebrew cask is available.

**Reply by ylluminate** [reply]
I wonder if using this in conjunction with Homebrew would be a good idea? Hmmm.

**Comment by olegshulyakov** [comment]
Here is a draft `open-webui.rb` Formula if you want to try:

```open-webui.rb
class OpenWebui < Formula
  include Language::Python::Virtualenv
  
  version "0.6.10"
  sha256 "0a7047f93e93207db4e6b75d40580ba9e98351c0dbdb2a532fc1b12ce5e06ead"
  
  url "https://files.pythonhosted.org/packages/source/o/open-webui/open_webui-#{version}.tar.gz"
  name "Open WebUI"
  desc "Open WebUI is an extensible, self-hosted interface for AI that adapts to your workflow, all while operating entirely offline; Supported LLM runners include Ollama and OpenAI-compatible APIs."
  homepage "https://github.com/open-webui/open-webui"
  
  auto_updates true
  depends_on "python@3.12"
    
  def install
    virtualenv_install_with_resources
  end
end
```

Here are the instructions how to add new one: https://docs.brew.sh/How-To-Open-a-Homebrew-Pull-Request

**Comment by cdamken** [comment]
# 🎉 Homebrew Formula Complete & Ready!

Hi everyone! Great news - I've built upon @olegshulyakov's excellent draft and created a **fully tested and working Homebrew formula** for Open WebUI!

## ✅ **Status: COMPLETED & THOROUGHLY TESTED**

The formula has been extensively tested and works perfectly:

- ✅ **Installation**: Successfully installs Open WebUI v0.6.14 (2.6GB, 80,500 files)
- ✅ **CLI Commands**: All functionality works (`open-webui --help`, `open-webui serve`, etc.)
- ✅ **Dependencies**: Automatically manages Python 3.12, Node.js 22, and ~98 Python packages
- ✅ **Uninstallation**: Clean removal of all files and dependencies
- ✅ **Homebrew Standards**: Follows all official Homebrew best practices
- ✅ **Automated Testing**: Comprehensive test suite included

## 📁 **Available Now**

The formula and supporting files are now available in the repository under `homebrew/`:

- **`homebrew/open-webui.rb`** - Main Homebrew formula
- **`homebrew/test-homebrew-formula.sh`** - Comprehensive test script  
- **`homebrew/README.md`** - Documentation and usage instructions

## 🚀 **Quick Start**

```bash
# Clone the repository
git clone https://github.com/open-webui/open-webui.git
cd open-webui/homebrew

# Install Open WebUI
brew install --formula ./open-webui.rb

# Verify installation
open-webui --help

# Start the server
open-webui serve
```

## 🧪 **Test It Yourself**

Want to verify it works? Run the included test suite:

```bash
cd homebrew
./test-homebrew-formula.sh
```

This will automatically test:
1. Formula syntax validation
2. Installation process
3. CLI functionality
4. Clean uninstallation  
5. Successful reinstallation

## 🔧 **Technical Details**

**What's Different from @olegshulyakov's Draft:**
- ✅ Updated to latest version (0.6.14)
- ✅ Fixed dependency management (all ~98 Python packages install correctly)
- ✅ Added Node.js 22 build dependency
- ✅ Uses proper virtual environment isolation
- ✅ Includes comprehensive testing
- ✅ Clean installation/uninstallation

**Key Features:**
- **Version**: 0.6.14
- **Dependencies**: Python 3.12 (runtime), Node.js 22 (build)
- **Size**: ~2.6GB installed
- **Python packages**: ~98 dependencies managed automatically
- **Environment**: Isolated Python virtual environment

## 🔄 **Next Steps**

1. **Pull Request**: Creating a PR to add this to the main repository
2. **Official Homebrew**: Once merged, this can be submitted to the official Homebrew repository  
3. **Future Updates**: The formula structure supports easy updates for new Open WebUI versions

## 🙏 **Credits**

Huge thanks to:
- @olegshulyakov for the initial draft that provided the foundation
- @Louis-Fiacre for starting this discussion
- Everyone who voted and commented - this shows there's real demand for this!

## 📋 **For Maintainers**

The formula is production-ready and follows all Homebrew conventions. It's been tested extensively and handles all the complex Python dependency management that Open WebUI requires.

---

**The Homebrew formula is now ready! No more workarounds needed - just `brew install` and go! 🍺**

Would love to hear feedback from anyone who tries it out!

**Reply by shrwnsan** [reply]
Awesome! Will wait for a brew install option

Any updates by the way?

(Tried cd open-webui/homebrew though not in latest repo)

**Comment by oxngon** [comment]
Bump. This needs to be merged into homebrew for simple `brew install` rather than git cloning locally

**Comment by w1nthinker** [comment]
rn via local, would appreciate open-webui on brew
